core/include/traccc/clusterization/impl/measurement_creation.ipp

```diff
@@ -19,41 +19,48 @@ inline scalar signal_cell_modelling(
 
 TRACCC_HOST_DEVICE
 inline vector2 position_from_cell(
-    const cell& cell,
+    const unsigned int cell_idx,
+    const edm::silicon_cell_collection::const_device& cells,
     const silicon_detector_description::const_device& det_descr) {
 
     // Retrieve the specific values based on module idx
-    const cell::link_type module_link = cell.module_link;
-    return {
-        det_descr.reference_x().at(module_link) +
-            (scalar{0.5} + cell.channel0) * det_descr.pitch_x().at(module_link),
-        det_descr.reference_y().at(module_link) +
-            (scalar{0.5} + cell.channel1) *
-                det_descr.pitch_y().at(module_link)};
+    const unsigned int module_idx = cells.module_index().at(cell_idx);
```

Comment:
I generally don't like that we don't have the cell type-defs anymore; can we put these in a helper struct or something?